GCRCatalogs/register.py

```diff
@@ -12,6 +13,21 @@
 _GITHUB_URL = 'https://raw.githubusercontent.com/LSSTDESC/gcr-catalogs/master/GCRCatalogs'
 _YAML_EXTENSIONS = ('.yaml', '.yml')
 
+# Keys appearing in yaml files whose values may be paths relative to _ROOT_DIR
+_PATH_LIKE = ('filename', 'addon_filename', 'base_dir', 'root_dir',
+              'catalog_root_dir', 'header_file', 'repo')
+
+def _get_rootdir():
+    with open(os.path.join(os.path.dirname(__file__),'site_config/site_rootdir.yaml')) as f:
+        d = yaml.safe_load(f)
+        host = socket.getfqdn()
+        for (k,v) in d.items():
+            if k in host:
+                return v
+        return None
+
+_ROOT_DIR = _get_rootdir()
```

Comment:
```suggestion
_DEFAULT_ROOT_DIR = _get_rootdir()
ROOT_DIR = _DEFAULT_ROOT_DIR

def reset_root_dir():
	ROOT_DIR = _DEFAULT_ROOT_DIR
```

Reply:
I like the idea of separating the root dir used from the default.  I'm not sure about exposing the variable, though.  If someone needs to override I'd rather they do it through a function call.

Reply:
Since there is really no real protected properties in Python, we do depend on  users being reasonable users. Hence, allowing ROOT_DIR to be a regular variable is a clear indication of that the users have control over it and they should be careful with it.

Reply:
Code should not depend on inputs being reasonable or users being careful.   GCR should check that it is reasonable.

Reply:
My point is that users can always edit `_DEFAULT_ROOT_DIR` or `_ROOT_DIR` even if there exists a function called `set_root_dir`. So I would prefer having fewer ways to do the same thing. We also need a `get_root_dir` if we use `_ROOT_DIR` instead of `ROOT_DIR`.

Reply:
Yes, people can modify variables, but, outside of personal scripts and interactive work, it's not a good thing.   I would rather provide a routine which can at least check that the value corresponds to an actual directory.    I don't mind also providing a one-line get routine.

Reply:
OK, so now there are two distinct questions: (1) the API for setting/getting `ROOT_DIR` and (2) whether `ROOT_DIR` should be checked to a valid path when set. 

What I have been discussing is (1). I prefer `GCRCatalogs.ROOT_DIR = ...` to `GCRCatalogs.set_root_dir(...)` for reasons I mentioned. But this is really just a question of API.

Question (2) is different --- I actually think it should _not_ be checked at set time. It could be checked later (as we discussed later). The reasons I think it shouldn't be checked at set time are:
- (a) `ROOT_DIR` is not used until the catalog is about to be loaded, and generally I don't want to prohibit the users to do whatever they want unless it is really going to cause an issue.
- (b) I can imagine cases where `ROOT_DIR`is set before the directory exist, but later at catalog load time, the directory becomes ready. 
- (c) I think `GCRCatalogs.set_root_dir(..)` should have the same behavior as `GCRCatalogs._ROOT_DIR = ...`. Otherwise users can just skip the API we design and opt to use `GCRCatalogs._ROOT_DIR`.

Reply:
My preference is to prevent users from stepping on their own toes where possible, when it doesn't entail any real loss of functionality. And also to catch errors earlier rather than later.  
 For a), something which is being done with no clear purpose in mind which could lead to failure later is better not done at all. For b) I don't think it's unreasonable to ask users to defer setting _ROOT_DIR until after the directory is created.
Concerning c), the whole reason for providing a function is that setting GCRCatalogs._ROOT_DIR directly really shouldn't be allowed.  It's unfortunate that Python **does** allow it and therefore some people may do it in inappropriate circumstances.  What we can do is provide a better interface, which is the one that is publicized.

Reply:
Right, so I think we do disagree on some fundamental principles here. For example, I think we should use example and documentation rather than hard-coded exceptions to prevent users from stepping on their own toes. Another difference is that I think while we can catch errors early, we should not raise exception until it is really necessary. 

So, generally I am ok to defer to the author's preference, but I also want to make sure philosophy is not completely inconsistent with other parts of the code.

I am ok with get/set API, but still would not want to raise exception if when `ROOT_DIR` is set to a non-existent path. Raising a warning is fine.

Reply:
It's now changed to warning.  Can we resolve this issue?